DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/ToHost/Geometry/PolycurveToHostConverter.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections.Generic;
+using System.Linq;
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Models;
+
+namespace Speckle.Converters.AutocadShared.ToHost.Geometry;
+
+/// <summary>
+/// A polycurve has segments as list and it can contain different kind of ICurve objects like Arc, Line, Polyline, Curve etc..
+/// If polycurve segments is consist of only with Arcs and Lines, it can be represented as Polyline in Autucad.
+/// Otherwise we convert it as spline (list of ADB.Entity) that switch cases according to each segment type.
+/// </summary>
+[NameAndRankValue(nameof(SOG.Polycurve), NameAndRankValueAttribute.SPECKLE_DEFAULT_RANK)]
+public class PolycurveToHostConverter : ISpeckleObjectToHostConversion
+{
+  private readonly IRawConversion<SOG.Polycurve, ADB.Polyline> _polylineConverter;
+  private readonly IRawConversion<SOG.Polycurve, List<ADB.Entity>> _splineConverter;
+
+  public PolycurveToHostConverter(
+    IRawConversion<SOG.Polycurve, ADB.Polyline> polylineConverter,
+    IRawConversion<SOG.Polycurve, List<ADB.Entity>> splineConverter
+  )
+  {
+    _polylineConverter = polylineConverter;
+    _splineConverter = splineConverter;
+  }
+
+  public object Convert(Base target)
+  {
+    SOG.Polycurve polycurve = (SOG.Polycurve)target;
+    bool convertAsSpline = polycurve.segments.Any(s => s is not SOG.Line and not SOG.Arc);
+    bool isPlanar = IsPolycurvePlanar(polycurve);
+
+    if (convertAsSpline || !isPlanar)
+    {
+      return _splineConverter.RawConvert(polycurve);
+    }
+    else
+    {
+      return _polylineConverter.RawConvert(polycurve);
+    }
+  }
+
+  private bool IsPolycurvePlanar(SOG.Polycurve polycurve)
+  {
+    double? z = null;
```

Comment:
Not urgent, but I wonder if there's a pattern here, maybe worth a POC: comment but I presume we just inherited this

Reply:
This might be removed completely when we have new `AutocadPolycurve` send